pkg/apis/openstackproviderconfig/v1alpha1/types.go

```diff
@@ -59,6 +59,9 @@ type OpenstackProviderSpec struct {
 	// The names of the security groups to assign to the instance
 	SecurityGroups []string `json:"securityGroups,omitempty"`
 
+	// The name of the secret containing the user data (startup script in most cases)
+	UserDataSecret string `json:"userDataSecret,omitempty"`
```

Comment:
Should we use a `SecretReference` of `k8s.io/api/core/v1` here? This would in the future give us the flexibility of defining the user-data per namespace. So in theory each cluster can have its own userData.

Reply:
I like the idea of using SecretReference. If `Namespace` is not set, we should default to the `machine.Namespace`, tho. thoughts?